lib/shelley/test/integration/Main.hs

```diff
@@ -162,13 +169,20 @@ main = withUtf8Encoding $ withTracers $ \tracers -> do
                 ByronTransactions.spec @n
                 ByronHWWallets.spec @n
                 Settings.spec @n
-            parallel $ describe "CLI Specifications" $ do
+
+            -- We run in to problems if we run CLI tests in parallel,
+            -- particularly in hydra. This is due to hpc causing multiple
+            -- processes to write to the same .tix file, or something.
+            parallelIf (not ci) $ describe "CLI Specifications" $ do
```

Comment:
Say this in the comment.

> Hydra run tests with code coverage enabled. CLI tests run multiple processes. These processes can try to write to the same .tix file simultaneously, causing errors.


We may be able to fix it by setting [`HPCTIXFILE`](https://downloads.haskell.org/~ghc/latest/docs/html/users_guide/profiling.html#envvar-HPCTIXFILE) before spawning the subprocesses, but that's probably something to try later.